python/foglamp/services/core/api/service.py

```diff
@@ -69,6 +69,50 @@ def get_service_records():
     return web.json_response(response)
 
 
+async def delete_service(request):
+    """ Delete an existing service
+
+    :Example:
+        curl -X DELETE http://localhost:8081/foglamp/service/<svc name>
+    """
+
+    try:
+        svc = request.match_info.get('service_name', None)
+
+        if svc is None or svc.strip() == '':
+            raise web.HTTPBadRequest(reason='Missing service_name in request URL')
+
+        # shutdown of service should remove it from service registry (as unregister)
+        # looks like a bug!
+        # python/foglamp/services/south/server.py:268
+        # check http://localhost:8081/foglamp/service
+
+        storage = connect.get_storage_async()
+
+        result = await get_schedule(storage, svc)
+        if result['count'] == 0:
+            raise web.HTTPBadRequest(reason='A service with this name does not exist.')
+
+        svc_schedule = result['rows'][0]
+
+        import uuid
+        sch_id = uuid.UUID(svc_schedule['id'])
+        if svc_schedule['enabled'].lower() == 't':
+            # disable it
+            await server.Server.scheduler.disable_schedule(sch_id)
+        # delete it
+        await server.Server.scheduler.delete_schedule(sch_id)
+
+        # delete config for the service name
+        await revert_configuration(storage, svc)  # Delete configuration entry
```

Comment:
We should also remove any child categories of the category whose name matches the service.